rest-adapter/src/main/java/uk/gov/justice/services/adapter/rest/envelope/RandomUUIDGenerator.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.services.adapter.rest.envelope;
+
+import java.util.UUID;
+
+/**
+ * Factory for creating new UUIDs.
+ */
+public class RandomUUIDGenerator {
+
+    /**
+     * Generate a random UUID.
+     * @return a UUID
+     */
+    public UUID generate() {
+        return UUID.randomUUID();
```

Comment:
(If you read the previous comment with an Australian accent, all of the question marks make sense?)